File: java/common-support/src/main/java/net/multi/language/common/message/event/order/OrderCreatedEvent.java
package net.multi.language.common.message.event.order;

import net.multi.language.common.marker.message.Channel;
import net.multi.language.common.marker.message.MessageDetails;
import net.multi.language.common.marker.message.Service;
import net.multi.language.common.message.AbstractDomainEvent;

@MessageDetails(
        publisher = Service.ORDER_SERVICE,
        subscribers = Service.BILLING_SERVICE,
        channel = Channel.ORDER_CREATED
)
public class OrderCreatedEvent extends AbstractDomainEvent {

    private static final String NAME = "OrderCreated";

    private final long customerId;
    private final long orderId;

    public OrderCreatedEvent(final long messageId, final long eventId, final long customerId, final long orderId) {
        super(messageId, eventId);
        this.customerId = customerId;
        this.orderId = orderId;
    }

    @Override
    public String getName() {
        return NAME;
    }

    public long getCustomerId() {
        return customerId;
    }

    public long getOrderId() {
        return orderId;
    }

    @Override
    public String toString() {
        return "customerId: " + customerId +
                ", orderId: " + orderId +
                '}';
    }
}

File: python/src/common/message/event/order/order_created_event.py
from typing import ClassVar

from common.message.abstract_domain_event import AbstractDomainEvent


class OrderCreatedEvent(AbstractDomainEvent):
    NAME: ClassVar[str] = "OrderCreated"

    customer_id: int
    order_id: int

    def get_name(self) -> str:
        return self.NAME

    def __str__(self) -> str:
        return f"customerId: {self.customer_id}, orderId: {self.order_id}"
